ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/CommandsValidator.scala

```diff
@@ -75,6 +74,29 @@ final class CommandsValidator(ledgerId: LedgerId) {
         ),
       )
 
+  private def maxTime(t1: Instant, t2: Instant): Instant =
+    if (t1.isAfter(t2)) t1 else t2
```

Comment:
You can `import Ordering.Implicits._` to then use `t1.max(t2)`.
This works because `Instant` is `Comparable[Instant]`. and `LowPriorityOrderingImplicits` defines an implicit `Ordering[A]` if there is an implicit `A => Comparable[A]`, which is trivially fulfilled for `Instant`.